plugins/org.polarsys.capella.scenario.editor.dsl/org.polarsys.capella.scenario.editor.dslscenario.ui/xtend-gen/org/polarsys/capella/scenario/editor/dslscenario/ui/contentassist/DslProposalProvider.java

```diff
@@ -95,18 +101,110 @@ public void getExistingParticipants(final String keyword, final ContentAssistCon
   public void completeSequenceMessage_Source(final EObject model, final Assignment assignment, final ContentAssistContext context, final ICompletionProposalAcceptor acceptor) {
     EList<Participant> _variablesDefinedBefore2 = this.variablesDefinedBefore2(((Model) model));
     for (final EObject el : _variablesDefinedBefore2) {
-      acceptor.accept(this.createCompletionProposal(((Actor) el).getName(), ((Actor) el).getName(), null, context));
+      this.createAcceptorForParticipants(el, acceptor, context);
     }
   }
   
   @Override
   public void completeSequenceMessage_Target(final EObject model, final Assignment assignment, final ContentAssistContext context, final ICompletionProposalAcceptor acceptor) {
     EList<Participant> _variablesDefinedBefore3 = this.variablesDefinedBefore3(((SequenceMessage) model));
     for (final EObject el : _variablesDefinedBefore3) {
-      acceptor.accept(this.createCompletionProposal(((Actor) el).getName(), ((Actor) el).getName(), null, context));
+      this.createAcceptorForParticipants(el, acceptor, context);
     }
   }
   
+  @Override
+  public void completeSequenceMessage_Name(final EObject model, final Assignment assignment, final ContentAssistContext context, final ICompletionProposalAcceptor acceptor) {
+    List<String> _messagesDefinedBefore = this.messagesDefinedBefore(((SequenceMessage) model));
+    for (final String el : _messagesDefinedBefore) {
+      acceptor.accept(this.createCompletionProposal((("\"" + el) + "\""), (("\"" + el) + "\""), null, context));
+    }
+  }
+  
+  public void createAcceptorForParticipants(final EObject el, final ICompletionProposalAcceptor acceptor, final ContentAssistContext context) {
```

Comment:
Simplyfy this function, no need for all the ifs.
Can use the Participant class.

Reply:
done